docs/proposal-0004-jetstream-preview.md

```diff
@@ -0,0 +1,118 @@
+
+Having access to experiments in preview would also help users to write and validate configs before the experiment has launched. Currently any pull-request adding a config for an experiment that isn't live cannot be merged or even be validated.
+
+## Risks
+
+There are some potential risks with providing users with the option of generating previews:
+* Users might optimize their configs based on the preview data which could impact the experiment results
+    * This is probably the biggest risk, however users might already do this, but instead rerun the entire experiment multiple times until they see the results they'd like.
+* Creating a preview might take too long
+    * Depending on the experiment size, and the provided preview parameters, generating a preview can take a while until results are available. Or in the worst case, if too much data is being processed the preview might not even finish when run locally.
+* With preview results being displayed in Looker instead of Experimenter it could be confusing for users to interpret the visualizations
+    * The Looker preview dashboard will show a subset of the results and without the explanations that are part of the Experimenter interface. Users that are less familiar with interpreting experiment data might be confused by that.
+
+## Alternatives considered
+
+### Airflow job for generating preview
+
+A new Airflow DAG for generating jetstream previews could be added. The DAG will have several jobs, similar to the [backfill DAG](https://workflow.telemetry.mozilla.org/dags/backfill/grid), that allow to set preview paramaters. Triggering the DAG will generate the preview datasets and output a link to the generated Looker dashboard. 
+Pro:
+* no need to install and run jetstream locally
+Con:
+* users might not be familiar with Airflow, and might be overwhelmed by what the different tasks mean and how to provide the right parameters.
+
+### Colab notebook for generating preview
+
+As long as the processed data is relatively small in size, jetstream should be able to run in a Colab notebook. A template could be provided to users and linked to from the CI where they can change the parameters and generate a preview.
+Pro:
+* no need to install and run jetstream locally
+Con:
+* overwhelming for less technical users
+
+### Job on cluster to generate preview
+
+A Google Cloud Function could be set up for users to trigger and provide parameters for generating the preview. The cloud function would create a new job on a Kubernetes cluster that runs the preview. This would be similar to triggering a rerun after a config change has been merged.
+Pro:
+* no need to install and run jetstream locally
+Con:
+* more complex setup
+* security concerns, who/when is this triggered. We don't want to trigger it for when it gets merged on `main`, but ideally as part of the pull-requests
```

Comment:
Seems like something we could setup with CircleCI triggers? Could we just have it run after validation? Or maybe if user wants to generate a preview they push to a branch like `preview-*`. Maybe I'm oversimplifying the concern, but seems like it could be a relatively low effort way to make this easier for lots of people.